src/editors/data/redux/thunkActions/app.js

```diff
@@ -41,6 +49,7 @@ export const initialize = (data) => (dispatch) => {
   dispatch(module.fetchUnit());
   dispatch(module.fetchStudioView());
   dispatch(module.fetchAssets());
+  dispatch(module.fetchCourseDetails());
```

Comment:
Its interesting how not all blocks might need all of these calls, but I'm struggling to find a way to specify a bunch of extra fetches depending on the type.

This might be the kind of time inheritance and overloading might have been very helpful.